Feature bjs2 96475 collaboration

samekeyev opened the pull request:
goal invitation - create, accept, reject, filter

Teymuruae committed:
feature-BJS2-96474-GoalsSetting [Постановка целей]

piligrim-curious committed:
Организация события

maxmerdock committed:
Выполнена задача "Покрыть код задачи "Подписка на других пользователей" unit-тестами"

DernovskiyAnton committed:
GlobalExceptionHandler не стал делать
пустые строки и импорты проверил
фильтрацию делал через Specification
Валидации сделал прямо в сервисе, не стал выносить в отдельный интерфейс

samekeyev committed:
BJS2-96475 collaborationgoal invitation - create, accept, reject, filter

samekeyev committed:
BJS2-96475 collaborationgoal invitation - create, accept, reject, filter

samekeyev committed:
BJS2-96475 collaboration
checkstyle

AndreySN7 commented on src/main/java/school/faang/user_service/service/goal/GoalInvitationServiceImpl.java:
```diff
@@ -0,0 +1,126 @@
+        }
+
+        int activeGoals = goalRepository.countActiveGoalsPerUser(invitedUserId);
+        if (activeGoals >= 3) {
```
Волшебное число "3". По условиям задачи рекомендуется вынести через аннотацию Value  в файл настроек. Либо как минимум его сделать через константу.

AndreySN7 commented on src/main/java/school/faang/user_service/service/goal/GoalInvitationServiceImpl.java:
```diff
@@ -0,0 +1,126 @@
+        GoalInvitation invitation = goalInvitationRepository.getByIdOrThrow(invitationId);
+
+        if (invitation.getStatus() == RequestStatus.ACCEPTED) {
+            throw new ForbiddenException("Приглашение уже принято");
```
Валидация юзера, это условие и следующее повторяются и в следующем методе. Я думаю стоит их вынести в отдельные блоки или в один блок и к нему уже обращаться. Если вынести в отдельные валидационные блоки каждое условие, мне кажется, проще будет юнит тесты писать в дальнейшем. Но это мое мнение, я могу ошибаться)

samekeyev committed:
Merge branch 'kraken-master-stream12' into feature-BJS2-96475-collaboration

AndreySN7 committed:
Решение задачи: Покрыть код задачи "Создание и настройка скиллов" unit-тестами

piligrim-curious committed:
[Организация события] * [Правки после кросс-ревью]

piligrim-curious committed:
[Организация события] * [Правки после кросс-ревью]

Teymuruae committed:
fixes after cross review

maxmerdock committed:
Правки по кросс-ревью

Teymuruae committed:
fixes after review

samekeyev committed:
BJS2-96475 collaboration
magic number

samekeyev committed:
Merge branch 'kraken-master-stream12' into feature-BJS2-96475-collaboration

DernovskiyAnton committed:
добавил логи
поменял исключения
добавил Transactional
поправил расположение пакетов

DernovskiyAnton committed:
добавил логи
поменял исключения
добавил Transactional
поправил расположение пакетов

AndreySN7 committed:
правки по ревью

Teymuruae committed:
BJS2-96517 Goals Setting unit tests done

DernovskiyAnton committed:
Добавил jacoco
Проверка покрытия на 70%
добавил переменную - потому что исключения используются в отчета и в покрытии

DernovskiyAnton committed:
Добавил jacoco
Проверка покрытия на 70%
добавил переменную - потому что исключения используются в отчета и в покрытии

SnurnicynKonstantin commented on src/main/java/school/faang/user_service/controller/goal/GoalInvitationController.java:
```diff
@@ -0,0 +1,31 @@
+    private final GoalInvitationService goalInvitationService;
+
+    GoalInvitationDto create(long goalId, CreateGoalInvitationDto invitationDto) {
+        if (invitationDto.getInvitedUserId() == null) {
```
Проверка полей, это логика приложения. Логику надо в сервисах писать. Контроллер должен быть максимально простым

SnurnicynKonstantin commented on src/main/java/school/faang/user_service/controller/goal/GoalInvitationController.java:
```diff
@@ -0,0 +1,31 @@
+    public void reject(long invitationId) {
+        goalInvitationService.reject(invitationId);
+    }
+
```
А где метод - просмотреть все приглашения с фильтрами ?

SnurnicynKonstantin commented on src/main/java/school/faang/user_service/dto/goal/GoalDto.java:
```diff
@@ -0,0 +1,14 @@
+
+@Getter
+@AllArgsConstructor
+public class GoalDto {
```
Не нашел использования этой ДТО. Значит надо ее удалить. + из маппера то же удалить

SnurnicynKonstantin commented on src/main/java/school/faang/user_service/mapper/GoalMapper.java:
```diff
@@ -0,0 +1,8 @@
+import school.faang.user_service.dto.goal.GoalDto;
+import school.faang.user_service.entity.goal.Goal;
+
+public interface GoalMapper {
```
Этот мапперне используется. Можем его удалить?

SnurnicynKonstantin commented on src/main/java/school/faang/user_service/service/goal/GoalInvitationServiceImpl.java:
```diff
@@ -0,0 +1,127 @@
+        invitation.setInvited(invited);
+        invitation.setStatus(RequestStatus.PENDING);
+
+        GoalInvitation saved = goalInvitationRepository.save(invitation);
```
`saved` -> `savedGoalInvitation`
Если бы было много кода, то в нем легко было бы запутаться, что такое saved)

SnurnicynKonstantin commented on src/main/java/school/faang/user_service/service/goal/GoalInvitationServiceImpl.java:
```diff
@@ -0,0 +1,127 @@
+    public void accept(long invitationId) {
+        GoalInvitation invitation = goalInvitationRepository.getByIdOrThrow(invitationId);
+
+        if (invitation.getStatus() == RequestStatus.ACCEPTED) {
```
А в требованиях есть что-то про проверку статуса? Если нет, то я предлагаю не усложнять и отказаться от проверки статуса

SnurnicynKonstantin commented on src/main/java/school/faang/user_service/service/goal/GoalInvitationServiceImpl.java:
```diff
@@ -0,0 +1,127 @@
+    public List<GoalInvitationDto> getByFilters(GoalInvitationFilterDto filters) {
+        List<GoalInvitation> invitations = goalInvitationRepository.findAll();
+
+        return invitations.stream()
```
Переделай пожалуйста фильтрацию так, как в этом уроке. Она по SOLID - https://faang-school.com/courses/4jnzmndg/9nzw6rrd

maxmerdock committed:
Выполнена задача "Swagger для UserService"

DernovskiyAnton committed:
second version

piligrim-curious committed:
[Организация события] * [Правки после ревью]

maxmerdock committed:
Правки по код-ревью

Teymuruae committed:
BJS2-96517 Goals Setting unit tests review fixes

Teymuruae committed:
BJS2-96643 get users controllers done

Teymuruae committed:
BJS2-96443 mentor request done

Teymuruae committed:
BJS2-96556 premium users search priority done

DernovskiyAnton committed:
Добавил шаги в ci

DernovskiyAnton committed:
Добавил шаги в ci

Teymuruae committed:
BJS2-96556 premium users search priority. Fixes after cross review

maxmerdock committed:
Правки по код-ревью

Teymuruae committed:
BJS2-96643 get users controllers review fixes

Teymuruae committed:
BJS2-96643 get users controllers review fixes

Teymuruae committed:
BJS2-96643 get users controllers review fixes

Teymuruae committed:
BJS2-96443 mentor request fixes after review

samekeyev committed:
BJS2-96487 user education

samekeyev committed:
Merge branch 'kraken-master-stream12' into feature-BJS2-96475-collaboration

samekeyev committed:
BJS2-96475 collaborationgoal invitation - create, accept, reject, filter
